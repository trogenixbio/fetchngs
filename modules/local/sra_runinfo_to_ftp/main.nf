
process SRA_RUNINFO_TO_FTP {

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    path runinfo

    output:
    //path "*.tsv"       , emit: tsv
    path "*.json"       , emit: json
    path "versions.yml", emit: versions
    // sra_runinfo_to_ftp.py \\
    //     ${runinfo.join(',')} \\
    //     ${runinfo.toString().tokenize(".")[0]}.runinfo_ftp.tsv
    script:
    """
    sra_runinfo_to_ftp_json.py \\
        ${runinfo.join(',')} \\
        ${runinfo.toString().tokenize(".")[0]}.runinfo_ftp.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
