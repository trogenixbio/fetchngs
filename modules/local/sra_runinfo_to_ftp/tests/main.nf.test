nextflow_process {

    name "Test process: SRA_RUNINFO_TO_FTP"
    script "../main.nf"
    process "SRA_RUNINFO_TO_FTP"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + 'GSE214215.runinfo.json', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
