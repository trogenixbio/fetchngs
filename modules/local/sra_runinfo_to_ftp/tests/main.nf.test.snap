{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "GSE214215.runinfo_ftp.json:md5,ddefdf7230e79e60e6afce91450ffb9a"
                ],
                "1": [
                    "versions.yml:md5,e95f8185f665127a73622a19d321bcca"
                ],
                "json": [
                    "GSE214215.runinfo_ftp.json:md5,ddefdf7230e79e60e6afce91450ffb9a"
                ],
                "versions": [
                    "versions.yml:md5,e95f8185f665127a73622a19d321bcca"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T15:12:27.58661199"
    }
}